database/database_test.go

```diff
@@ -234,3 +234,180 @@ func TestStatesWithStatus(t *testing.T) {
 		}
 	}
 }
+
+// setTests is a struct that is used in testing SetIneligibleDomainStates
+var setTests = []struct {
+	description       string
+	domainStates      []IneligibleDomainState
+	wantStatusReports []string
+}{
+	{
+		"two domains",
+		[]IneligibleDomainState{
+			{Name: "youtube.test", Policy: "bulk-1-year"},
+			{Name: "garron.test", Policy: "bulk-1-year"},
+		},
+		[]string{"Updating 2 entries...", " done.\n"},
+	},
+	{
+		"bulk-18-week",
+		[]IneligibleDomainState{{Name: "gmail.test", Policy: "bulk-18-week"}},
+		[]string{"Updating 1 entries...", " done.\n"},
+	},
+	{
+		"bulk-1-year",
+		[]IneligibleDomainState{{Name: "wikipedia.test", Policy: "bulk-1-year"}},
+		[]string{"Updating 1 entries...", " done.\n"},
+	},
+}
+
+// Test SetIneligibleDomainStates
+func TestSetIneligibleDomainStates(t *testing.T) {
+	resetDB()
+
+	for _, tt := range setTests {
+
+		var statuses []string
+		statusReport := func(format string, args ...interface{}) {
+			formatted := fmt.Sprintf(format, args...)
+			statuses = append(statuses, formatted)
+		}
+
+		err := testDB.SetIneligibleDomainStates(
+			tt.domainStates,
+			statusReport,
+		)
+		if err != nil {
+			t.Errorf("[%s] cannot put states %s", tt.description, err)
+			return
+		}
+
+		if !reflect.DeepEqual(statuses, tt.wantStatusReports) {
+			t.Errorf("[%s] Incorrect status reports: %#v", tt.description, statuses)
+		}
+	}
+}
+
+// getAndDeleteTests is a struct that is used in testing GetIneligibleDomainStates
+// and DeleteIneligibleDomainStates
+var getAndDeleteTests = []struct {
+	description  string
+	domainStates []string
```

Comment:
I'd expect a field with a name `domainStates` to have a type like `[]DomainState` or `[]IneligibleDomainState`. Perhaps this should be `domains` or `domainNames` instead?